astropy/coordinates/representation.py

```diff
@@ -114,9 +119,22 @@ def get_name(cls):
             name = name[:-14]
         return name
 
-    def __getitem__(self, view):
-        return self.__class__(*[getattr(self, component)[view]
-                                for component in self.components])
+    def _replicate(self, method, *args, **kwargs):
```

Comment:
Thinking about how this might be applied in future, is "replicate" the best name for this function? To me, it implies that the returned ```representation``` will be the same as ```self```, but in fact it is often transformed: for example if the method is ```reshape``` or ```sum``` then the returned ```representation``` is a different shape to the original.